### ISSUE TITLE
[terra-form-select] Cannot select ClearOption via keyboard navigation

### ISSUE THREAD

**Issue by dev-cprice** [cerner/terra-core#2661]
# Bug Report

## Description

Using keyboard navigation to unselect an option on [terra-form-select examples](https://engineering.cerner.com/terra-ui/components/terra-form-select/form-select/select) sets the active element to the `<ClearOption>`, but presses `ENTER` does nothing

## Steps to Reproduce

1. Navigate to [deployed terra ui site](https://engineering.cerner.com/terra-ui/components/terra-form-select/form-select/select)
2. Scroll down to the clear option example
3. Select an option
4. Using the keyboard, navigate to the clear option
5. Press enter, nothing happens
6. Click on the clear option, select is cleared

## Expected Behavior

When using keyboard navigation, the clear option is selectable using `ENTER`

## Possible Solution

For `Combobox`:

When the active option is the `<ClearOption>`, it's falsey since it's set to the clear option's `value`, which is `""` (empty string). So, this [line](https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/combobox/Menu.jsx#L306) just needs to change from a truthiness check of `active` to `active !== null`

For `SingleSelect`:

Same as above, this [line](https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/single/Menu.jsx#L300)

ArrowUp is also not able to find the clear option for single select variant. This is because the active option isn't being set, similar reason (`falsey` check instead of `!== null`) [in getActiveOptionFromProps here](https://github.com/cerner/terra-core/blob/master/packages/terra-form-select/src/single/Menu.jsx#L77)

## Environment

* Component Name and Version: terra-form-select latest
